### ISSUE TITLE
[.xts segfaults if numeric or logical j contains NA

### ISSUE THREAD

**Issue by IlyaKipnis** [joshuaulrich/xts#97]
A `NA` in a numeric or logical `j` will cause a segfault if `[.xts` is called on an xts object with more than one column.

```r
require(xts)
.xts(1,1)[,NA]  # doesn't crash; error could be more informative though
.xts(t(c(1,1)),1)[,NA]  # crashes
.xts(t(c(1,1)),1)[,c(1,NA)]  # crashes
```

-Ilya

**Comment by joshuaulrich** [comment]
This worked as of [cfafaaa](https://github.com/joshuaulrich/xts/commit/cfafaaac2f169c8191518e1cef380055b18387c6).  It looks like the move to `_do_subset_xts` and `extract_col` created regression bugs.
```r
.xts(t(c(1,1)),1)[,NA]  # crashes in .Call("extract_col",...)
.xts(t(c(1,1)),1)[1,NA]  # crashes in .Call("_do_subset_xts",...)
```

**Comment by joshuaulrich** [comment]
All 3 issues have been fixed.

1. The crash caused by `.xts(t(c(1,1)),1)[,NA]` is fixed in 724660aaf7446ecfeee34e07597fdbb06c036fe0.
2. The crash caused by `.xts(t(c(1,1)),1)[1,NA]` is fixed in fa1c82f189cdd95638c4129ea7000a919835f906.
3. The error caused by `.xts(1,1)[,NA]` is fixed in 0cb64b388864eff3f7ef0e53f6683d3f7a3cce35.
